lightning/src/ln/channel.rs

```diff
@@ -795,6 +791,71 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		Ok(chan)
 	}
 
+	pub(super) fn monitor_would_broadcast_at_height<L: Deref>(&self, height: u32, logger: &L) -> bool where L::Target: Logger {
+		macro_rules! add_htlc_output {
+			($htlc: expr, $offered: expr, $list: expr) => {
+				$list.push(HTLCOutputInCommitment{
+					offered: $offered,
+					amount_msat: $htlc.amount_msat,
+					cltv_expiry: $htlc.cltv_expiry,
+					payment_hash: $htlc.payment_hash,
+					transaction_output_index: None
+				});
+			}
+		}
+
+		let cap = self.pending_inbound_htlcs.len() + self.pending_outbound_htlcs.len();
+		let mut remote_outputs = Vec::with_capacity(cap);
+		let mut local_outputs = Vec::with_capacity(cap);
+		let awaiting_raa = (self.channel_state & ChannelState::AwaitingRemoteRevoke as u32) != 0;
+		for ref htlc in self.pending_inbound_htlcs.iter() {
```

Comment:
I figure you should be able to do this with an `iter().map(..)` to avoid putting every HTLC into a Vec here, no?